### DISCUSSION TITLE
Provide option to time delta on upstream tasks in TimeDeltaSensor

### DISCUSSION THREAD

**Discussion by klsnreddy** [apache/airflow#29646] [General]
**Description**
Provide option to time delta on upstream tasks in TimeDeltaSensor. Currently it adds delta to the schedule end time, irrespective of where the TimeDeltaSensor is in DAG.

**Use case / motivation**
We have a requirement of sending an email after 30 mins of previous task completion, but TimeDeltaSensor only waits till 30 mins after schedule end time, and by the the time we reach sensor if that delta + end time is less then current time, it will continue the execution with out waiting. Even this feature would be useful when working with eventual consistent system, where you want to wait for couple of minutes before proceeding further.

**Comment by ceversdyk** [comment]
I would love to see this feature implemented. Currently trying to think of a good workaround, maybe have the upstream task push a timestamp to xcom right before it finishes execution? Or a custom Sensor which checks the database for upstream task completion time? Yes, it would be much more helpful if the TimeDeltaSensor could reference the actual completion time of an upstream dependent task.

**Comment by potiuk** [comment]
This will be much better done with custom  `SendEmailAfterSomeTime` operator that would first  defer itself with TimeDeltaTrigger and run "email_send" after trigger fires. 

This would be much better solution:

* you would see that the task is deferred while waiting
* it would take practically no resources when waiting
* you just need single taks rather than separate TimeDeltaSensor and SendEmail

Converting to discussion - I firmly believe Deferrable Operator with TimeDeltaTrigger is much better solution for that but if there are some other comments on that we can continue discussing it.
